src/utils/solanaUtils.ts

```diff
@@ -513,3 +670,131 @@ export const groupCollectablesWithMetaData = (collectables: Collectable[]) => {
 
   return collectablesGroupedByName
 }
+
+/**
+ *
+ * @param mint mint address
+ * @param metaplex metaplex connection
+ * @returns collectable
+ */
+export const getCollectableByMint = async (
+  mint: web3.PublicKey,
+  metaplex: Metaplex,
+): Promise<Collectable | null> => {
+  try {
+    const collectable = await metaplex.nfts().findByMint({ mintAddress: mint })
+    if (!collectable.json && collectable.uri) {
+      const json = await (await fetch(collectable.uri)).json()
+      return { ...collectable, json }
+    }
+    return collectable
+  } catch (e) {
+    return null
+  }
+}
+
+/**
+ *
+ * @param address public key of the account
+ * @param cluster cluster
+ * @param oldestTransaction starting point of the transaction history
+ * @returns transaction history
+ */
+export const getAllTransactions = async (
+  address: string,
+  cluster: web3.Cluster,
+  oldestTransaction?: string,
+): Promise<(EnrichedTransaction | web3.ConfirmedSignatureInfo)[]> => {
+  const pubKey = new web3.PublicKey(address)
+  const conn = getConnection(cluster)
+  const metaplex = new Metaplex(conn, { cluster })
+  const parseTransactionsUrl = `${Config.HELIUS_API_URL}/v0/transactions/?api-key=${Config.HELIUS_API_KEY}`
+
+  try {
+    const txList = await conn.getSignaturesForAddress(pubKey, {
+      before: oldestTransaction,
+      limit: 100,
+    })
+    const sigList = txList.map((tx) => tx.signature)
+
+    if (cluster !== 'mainnet-beta') {
+      return txList
+    }
+
+    const { data } = await axios.post(parseTransactionsUrl, {
+      transactions: sigList,
+    })
+
+    /*
+     * TODO: Remove this once helius nft indexer is live
```

Comment:
when will it be live?

Reply:
No idea but I can try to find out with the helius team